src/Akka.Hosting/HealthChecks/AkkaHealthCheckExtensions.cs

```diff
@@ -17,7 +17,7 @@ public static HealthCheckRegistration ToHealthCheckRegistration(this AkkaHealthC
     {
         // func for lazily instantiating the health check registration
         Func<IServiceProvider, IHealthCheck> adapter = provider =>
-            new HealthCheckAdapter(registration.HealthCheck, provider.GetRequiredService<ActorSystem>());
+            new HealthCheckAdapter(registration.Factory(provider), provider.GetRequiredService<ActorSystem>());
```

Comment:
Changed the health check adapter only slightly here - it's still a delayed instantiation until the health checks actually get invoked.